### DISCUSSION TITLE
please Explane

### DISCUSSION THREAD

**Discussion by kebab03** [ManimCommunity/manim#2906] [General]
what is  representing x  _______in lambda x: x.become(Angle(line1, line_moving, radius=0.5, other_angle=False)  .....can someone teach me .Thank you in Advance.

in the following code of manim 

class MovingAngle(Scene):
    def construct(self):
        rotation_center = LEFT

        theta_tracker = ValueTracker(110)
        line1 = Line(LEFT, RIGHT)
        line_moving = Line(LEFT, RIGHT)
        line_ref = line_moving.copy()
        line_moving.rotate(
            theta_tracker.get_value() * DEGREES, about_point=rotation_center
        )
        a = Angle(line1, line_moving, radius=0.5, other_angle=False)
        tex = MathTex(r"\theta").move_to(
            Angle(
                line1, line_moving, radius=0.5 + 3 * SMALL_BUFF, other_angle=False
            ).point_from_proportion(0.5)
        )

        self.add(line1, line_moving, a, tex)
        self.wait()

        line_moving.add_updater(
            lambda x: x.become(line_ref.copy()).rotate(
                theta_tracker.get_value() * DEGREES, about_point=rotation_center
            )
        )

        a.add_updater(
            lambda x: x.become(Angle(line1, line_moving, radius=0.5, other_angle=False))
        )
        tex.add_updater(
            lambda x: x.move_to(
                Angle(
                    line1, line_moving, radius=0.5 + 3 * SMALL_BUFF, other_angle=False
                ).point_from_proportion(0.5)
            )
        )

        self.play(theta_tracker.animate.set_value(40))
        self.play(theta_tracker.animate.increment_value(140))
        self.play(tex.animate.set_color(RED), run_time=0.5)
        self.play(theta_tracker.animate.set_value(350))

**Comment by SuperUser-Vermeylen** [comment]
This is a good explanation of [lambda](https://www.w3schools.com/python/python_lambda.asp).

For your case, x is going to be an object and it appears the example is using lambda as a transformation.
Is this your code or an example from Manim?
Can you post the Manim link, if it is an example?

**Reply by kebab03** [reply]
An example from manim docs web.

**Reply by kebab03** [reply]
https://docs.manim.community/en/stable/examples.html    scrolldown or  use find to go to the MovingAngle class

**Reply by SuperUser-Vermeylen** [reply]
I'm pretty novice to coding, but I think that lambda is a convenient way to write a function without having to call a "def." You can nest a lambda inside of another "def."
In the example x can be any mobejct (I think) that you want transformed into the line.
So it seems that consecutive lines are being transformed into one another through the method(?) "become" in the lambda function.
I think animate calls the updater functions as the scene progresses during rendering.
